util/module_logger_test.go

```diff
@@ -0,0 +1,258 @@
+	l.Errorf("Error Format", 1, 2)
+	suite.assertMsgf(ErrorLevel, "Error Format", 1, 2)
+	l.Fatal("Fatal Msg", 1, 2)
+	suite.assertMsg(FatalLevel, "Fatal Msg", 1, 2)
+	l.Fatalf("Fatal Format", 1, 2)
+	suite.assertMsgf(FatalLevel, "Fatal Format", 1, 2)
+	l.Panic("Panic Msg", 1, 2)
+	suite.assertMsg(PanicLevel, "Panic Msg", 1, 2)
+	l.Panicf("Panic Format", 1, 2)
+	suite.assertMsgf(PanicLevel, "Panic Format", 1, 2)
+}
+
+func (suite *ModuleTestSuite) TestPanic() {
+	l := suite.ml.GetLogger("testpanic")
+	l.Debug("Debug Msg", 1, 2)
+	suite.assertNoMsg()
+	l.Debugf("Debug Format", 1, 2)
+	suite.assertNoMsg()
+	l.Info("Info Msg", 1, 2)
+	suite.assertNoMsg()
+	l.Infof("Info Format", 1, 2)
+	suite.assertNoMsg()
+	l.Warn("Warn Msg", 1, 2)
+	suite.assertNoMsg()
+	l.Warnf("Warn Format", 1, 2)
+	suite.assertNoMsg()
+	l.Error("Error Msg", 1, 2)
+	suite.assertNoMsg()
+	l.Errorf("Error Format", 1, 2)
+	suite.assertNoMsg()
+	l.Fatal("Fatal Msg", 1, 2)
+	suite.assertNoMsg()
+	l.Fatalf("Fatal Format", 1, 2)
+	suite.assertNoMsg()
+	l.Panic("Panic Msg", 1, 2)
+	suite.assertMsg(PanicLevel, "Panic Msg", 1, 2)
+	l.Panicf("Panic Format", 1, 2)
+	suite.assertMsgf(PanicLevel, "Panic Format", 1, 2)
+}
+
+func (suite *ModuleTestSuite) TestOff() {
+	l := suite.ml.GetLogger("testoff")
+	l.Panic("Panic Msg", 1, 2)
+	suite.assertNoMsg()
+}
+
+func (suite *ModuleTestSuite) TestChangeLogger() {
+	newLogger := &dummyLogger{}
+	l := suite.ml.SetLogger(newLogger).GetLogger("testpanic")
+	l.Fatal("Fatal Msg", 1, 2)
+	newLogger.assertNoMsg(suite.T())
+	l.Panic("Panic Msg", 1, 2)
+	newLogger.assertMsg(suite.T(), PanicLevel, "Panic Msg", 1, 2)
+}
+
+func (suite *ModuleTestSuite) TestCache() {
+	suite.ml.SetModuleLevel("m1", InfoLevel)
+	suite.ml.SetModuleLevel("m2", InfoLevel)
+	_ = suite.ml.GetLogger("m1")
+	m2 := suite.ml.GetLogger("m2")
+	m2.Debug("m2")
+	suite.assertNoMsg()
+	m2.Info("m2")
+	suite.assertMsg(InfoLevel, "m2")
+}
+
+// Make sure the logger returned by WithField is still wrapped.
+func (suite *ModuleTestSuite) TestWithField() {
+	l := suite.ml.GetLogger("testpanic")
+	new_l := l.WithField("field", 1)
+	assert.Equal(suite.T(), suite.dl.fieldKey, "field")
+	assert.Equal(suite.T(), suite.dl.fieldValue, 1)
+	new_l.Fatal("Fatal Msg", 1, 2)
+	suite.assertNoMsg()
+	new_l.Panic("Panic Msg", 1, 2)
+	suite.assertMsg(PanicLevel, "Panic Msg", 1, 2)
+}
+
+type dummyLogFields struct{}
+
+func (dummyLogFields) Fields() map[string]interface{} {
+	return nil
+}
+
+// Make sure the logger returned by WithFields is still wrapped.
+func (suite *ModuleTestSuite) TestWithFields() {
+	l := suite.ml.GetLogger("testpanic")
+	dlf := new(dummyLogFields)
+	new_l := l.WithFields(dlf)
+	assert.Equal(suite.T(), suite.dl.fields, dlf)
+	new_l.Fatal("Fatal Msg", 1, 2)
+	suite.assertNoMsg()
+	new_l.Panic("Panic Msg", 1, 2)
+	suite.assertMsg(PanicLevel, "Panic Msg", 1, 2)
+}
+
+func (suite *ModuleTestSuite) TestPreconditions() {
+	assert := assert.New(suite.T())
```

Comment:
The Suite struct embeds `*assert.Assertions` and this gives access to all of its methods. I advise you to change:

```
assert.NoError
assert.Error
```

to 

```
s.NoError
s.Error
```
```